components/properties/src/props.rs

```diff
@@ -2455,3 +2502,136 @@ impl_value_getter! {
         pub fn get_enum_to_long_name_mapper() / enum_to_long_name_mapper() -> PropertyEnumToValueNameLinearMapper / PropertyEnumToValueNameLinearMapperBorrowed;
     }
 }
+#[cfg(test)]
+mod test_enumerated_property_completeness {
+    use super::*;
+    use alloc::collections::BTreeSet;
+    use std::cmp::Ordering;
+    use std::fmt;
+    enum Origin {
+        Data,
+        Consts,
+    }
+    struct NameValue {
+        origin: Origin,
+        name: String,
+        value: u16,
+    }
+
+    impl fmt::Debug for NameValue {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> fmt::Result {
+            let origin = match self.origin {
+                Origin::Data => "Data",
+                Origin::Consts => "Consts",
+            };
+            write!(f, "{}:\t{:?}:\t{:?}", origin, self.name, self.value)
+        }
+    }
+
+    impl PartialOrd for NameValue {
+        fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+            Some(self.value.cmp(&other.value))
+        }
+    }
+
+    impl PartialEq for NameValue {
+        fn eq(&self, other: &Self) -> bool {
+            self.value == other.value
+        }
+    }
+
+    impl Eq for NameValue {}
+
+    impl Ord for NameValue {
+        fn cmp(&self, other: &Self) -> Ordering {
+            self.value.cmp(&other.value)
+        }
+    }
+
+    fn check_enum<'a>(
+        lookup: &PropertyValueNameToEnumMapV1<'static>,
+        consts: impl IntoIterator<Item = &'a (&'static str, u16)>,
+    ) {
+        let data = lookup
+            .map
+            .iter_copied_values()
+            .map(|(name, value)| NameValue {
+                origin: Origin::Data,
+                name: String::from_utf8(name.as_byte_slice().to_vec()).unwrap(),
+                value,
+            })
+            .collect::<BTreeSet<NameValue>>();
+        let consts = consts
+            .into_iter()
+            .map(|(name, value)| NameValue {
+                origin: Origin::Consts,
+                name: name.to_string(),
+                value: *value,
+            })
+            .collect::<BTreeSet<NameValue>>();
+
+        let diff: Vec<_> = data.symmetric_difference(&consts).collect();
```

Comment:
Suggestion (optional): another way to do this, which would avoid the custom `Eq` and `Ord` impls, and reduce the amount of code overall, would be

```rust
let mut map: BTreeMap<u16, (Origin, String)> = /* data collection */;
let mut diff: Vec<(u16, (Origin, String))> = Vec::new();
for (origin, name, value) in /* consts iterator */ {
    if map.remove(origin).is_none() {
        diff.push((value, (origin, name)));
    }
}
diff.extend(map.into_iter());
```

Reply:
That's much cleaner. Thanks for the suggestion!